op-node/rollup/derive/system_config.go

```diff
@@ -69,50 +69,115 @@ func ProcessSystemConfigUpdateLogEvent(destSysCfg *eth.SystemConfig, ev *types.L
 	}
 	// indexed 1
 	updateType := ev.Topics[2]
-	// unindexed data
+
+	// Create a reader of the unindexed data
+	reader := bytes.NewReader(ev.Data)
+	// Allocate a placeholder word to read into
+	word := make([]byte, 32)
```

Comment:
Maybe it's cleaner to define an inline function that reads a word as 32 byte array, instead of repeating the same snippet of reading and converting it?
And the length checks of data should prevent reading errors already, so the inline function can maybe omit the error (although still nice if we can type check or detect programming errors there somehow)

Reply:
Good suggestion on the inline func. Updated- let me know if this was what you were looking for here.